test/unit_test/character_sheet_unit_tests/FocusesSetUnitTests.java

```diff
@@ -0,0 +1,101 @@
+package unit_test.character_sheet_unit_tests;
+
+import static org.junit.Assert.*;
+
+import java.security.InvalidParameterException;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import rpg_database.character_sheet.CharacterSheet;
+import rpg_database.character_sheet.Fields;
+import rpg_database.character_sheet.Focuses;
+import rpg_database.character_sheet.Focus;
+import rpg_database.character_sheet.FocusesSet;
+
+public class FocusesSetUnitTests {
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	@Test
+	public void expectException_SetFocusesMalformedInput() {
+		expectExceptionWithMessage(InvalidParameterException.class,
+				"class java.lang.String value is not an instance of class rpg_database.character_sheet.Focuses");
+		final String malformedInput = "MALFORMED INPUT";
+		CharacterSheet characterSheet = new CharacterSheet("CharacterSheet");
+		characterSheet.setData(Fields.FOCUSES, malformedInput);
+	}
+
+	// functional tests
+	@Test
+	public void testSetFocusWithFocusesSet() {
+		CharacterSheet characterSheet = new CharacterSheet("TestCharacterSheet");
+		FocusesSet expectedFocuses = new FocusesSet(new Focus(Focuses.ETIQUETTE));
+		characterSheet.setData(Fields.FOCUSES, expectedFocuses);
+		FocusesSet actualFocus = characterSheet.getData(Fields.FOCUSES);
+		assertEquals(expectedFocuses, actualFocus);
+	}
+
+	@Test
+	public void testSetMoreDifferentFocus() {
+		CharacterSheet characterSheet = new CharacterSheet("TestCharacterSheet");
+		FocusesSet expectedFocuses = new FocusesSet(new Focus(Focuses.ANIMAL_HANDLING), new Focus(Focuses.ARCANE_LANCE), new Focus(Focuses.BRAWLING));
+		characterSheet.setData(Fields.FOCUSES, expectedFocuses);
+		FocusesSet actualFocuses = characterSheet.getData(Fields.FOCUSES);
+		assertEquals(expectedFocuses, actualFocuses);
+	}
+
+	@Test
+	public void testGetSingleTimeSetIsFocusImproved() {
+		CharacterSheet characterSheet = new CharacterSheet("TestCharacterSheet");

```

Comment:
Miért kell ehhez a CharacterSheet? FocusesSetből nem elég tesztelni?

És miért kell Focus objektumokat átadni a Setnek? Miért nem lehet csak szimplán Focuses enumokat átadni neki, aztán majd ő magában kitalálja hogy hogyan tárolja le azt ha több ugyanolyan van átadva. Ezt lehetne elegánsabban is mint minden paraméterezést kivezetni a legkisebb ellenállás szintjére. Persze kell hozzá kis extra exception handling olyan esetekre amikor a paraméterlista nem feldolgozható, meg az add-et meg addAll-t nem lehet defaultba hagyni de meló nélkül nem lesz okosabb a set....